### PR TITLE
turbo: 1.11.3 -> 1.13.2

### PR DESCRIPTION
**Pull request by samhh**
## Description of changes

Here's the changelog for 1.13.2, though we've skipped quite a few releases: https://github.com/vercel/turbo/releases/tag/v1.13.2

We may also consider setting the following [env vars](https://turbo.build/repo/docs/reference/system-variables#system-environment-variables), the latter of which is new:

- `$TURBO_NO_UPDATE_NOTIFIER`
- `$TURBO_TELEMETRY_DISABLED` or `$TURBO_TELEMETRY_MESSAGE_DISABLED`

I don't know what the nixpkgs position is on stuff like this so for now the PR only upgrades the package.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pbsds — APPROVED** [review]
Several packages disable auto updates, some disable telemetry. 
I don't think we have a set policy on it, what would you prefer?

**Comment by samhh** [comment]
> Several packages disable auto updates, some disable telemetry. I don't think we have a set policy on it, what would you prefer?

Personally I'd prefer to disable both primarily to minimise noise, not only in dev environments where you might not see the telemetry message often but also in CI.

**Comment by pbsds** [comment]
Then by all means go ahead. I assume these must be set at runtime, meaning you have to wrap the binary

**Comment by samhh** [comment]
I've copied the approach taken with GNU parallel, splitting Turbo into `turbo` and `turbo-unwrapped`. `turbo` now disables telemetry and the update notifier by default, though each can be overridden.  The telemetry change at least is easy to validate:

```console
$ nix run ".#turbo-unwrapped" -- telemetry status
Status: Enabled
$ nix run ".#turbo" -- telemetry status
Status: Disabled
```

File: pkgs/tools/misc/turbo/wrapper.nix [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{ lib, runCommand, makeWrapper, turbo-unwrapped
+, disableTelemetry ? true, disableUpdateNotifier ? true }:
+
+runCommand "turbo" { nativeBuildInputs = [ makeWrapper ]; } ''
```
Comment (pbsds) [inline pkgs/tools/misc/turbo/wrapper.nix:4]: I assume `runCommand` supports these inputs.

```suggestion
runCommandNoCC "turbo-${version}" { 
  inherit (turbo-unwrapped) version meta; 
  pname = "turbo";
  nativeBuildInputs = [ makeWrapper ]; 
} ''
```

**Review by pbsds — COMMENTED** [review]
The by-name check i'd say hit a snag in this case and can be ignored, as this package is not trivially migrated to by-name. 
CC @infinisil:  https://github.com/NixOS/nixpkgs/actions/runs/8908188109/job/24463383639?pr=301781 caused by https://github.com/NixOS/nixpkgs/pull/301781/commits/2c0219bb232c267fda35765954d0ca5df5fa3b08

This wrapping scheme fails to propagate `version` and `meta`, please try the following

**Comment by pbsds** [comment]
actually, https://github.com/NixOS/nixpkgs/pull/304018 has a nice approach you may use for inspiration

**Pull request merged**